packages/expo-cli/src/appleApi/authenticate.ts

```diff
@@ -7,6 +11,99 @@ import prompt from '../prompt';
 import { nonEmptyInput } from '../validators';
 import { runAction, travelingFastlane } from './fastlane';
 
+const NO_STORE_PASSWORD = getenv.boolish('EXPO_NO_STORE_PASSWORD', false);
+
+function getKeychainServiceName(appleId: string): string {
+  return `deliver.${appleId}`;
+}
+
+async function deletePasswordAsync({
+  appleId,
+}: Pick<AppleCredentials, 'appleId'>): Promise<boolean> {
+  // Mac only right now.
+  if (process.platform !== 'darwin') {
+    return false;
+  }
+  const keychainService = getKeychainServiceName(appleId);
+  return new Promise((resolve, reject) => {
+    keychain.deletePassword(
+      { account: appleId, service: keychainService, type: 'internet' },
+      (error: Error) => {
+        if (error) {
+          if (error.message.match(/Could not find password/)) {
+            return resolve(false);
+          }
+          reject(error);
+        } else {
+          log('Removed Apple ID password from the native key chain.');
+          resolve(true);
+        }
+      }
+    );
+  });
+}
+
+async function getPasswordAsync({
+  appleId,
+}: Pick<AppleCredentials, 'appleId'>): Promise<string | null> {
+  // Mac only right now.
+  if (process.platform !== 'darwin') {
+    return null;
+  }
+  // If the user opts out, delete the password.
+  if (NO_STORE_PASSWORD) {
+    await deletePasswordAsync({ appleId });
+    return null;
+  }
+
+  const keychainService = getKeychainServiceName(appleId);
+  return new Promise((resolve, reject) => {
+    keychain.getPassword(
+      { account: appleId, service: keychainService, type: 'internet' },
+      (error: Error, password: string) => {
+        if (error) {
+          if (error.message.match(/Could not find password/)) {
+            return resolve(null);
+          }
+          reject(error);
+        } else {
+          resolve(password);
+        }
+      }
+    );
+  });
+}
+
+async function storePasswordAsync({
+  appleId,
+  appleIdPassword,
+}: AppleCredentials): Promise<boolean> {
+  // Mac only right now.
+  if (process.platform !== 'darwin') {
+    return false;
+  }
+  if (NO_STORE_PASSWORD) {
+    log('Skip storing Apple ID password in the native key chain.');
+    return false;
+  }
+  log(
+    'Saving Apple ID password to the local native key chain. You can disable this with `EXPO_NO_STORE_PASSWORD=true`'
```

Comment:
If someone didn't want their password stored in the first place, the next question they'd have is 'How do I erase the password that is stored in keychain now?'. Consider making an `expo.fyi` link for more info and instructions on how to do this